spec/support/ar_classes.rb

```diff
@@ -1,12 +1,35 @@
-ActiveRecord::Base.establish_connection(
-  :adapter => 'sqlite3',
-  :database => ':memory:'
-)
+def establish_active_record_connection
+  ActiveRecord::Base.establish_connection(
+    :adapter => 'sqlite3',
+    :database => ':memory:'
+  )
+
+  [:MockableModel,
+   :SubMockableModel,
+   :AssociatedModel,
+   :AlternatePrimaryKeyModel].each do |host|
+    next unless Kernel.const_defined? host
+    Connections.extended(Kernel.const_get(host))
+  end
+end
```

Comment:
Why?

Reply:
See the spec. 
```ruby
    context "without database available" do
      before { clear_active_record_connection }
       after { establish_active_record_connection }
       context "with use_active_record set to true" do
```
I wrote a spec which tests exactly this behavior.

`establish_active_record_connection` takes care of creating AR connection, but when it run for the first time the models are not yet defined and later in the middle of the suite the connection is dropped which leads to the fact that the db tables are missing and need to be migrated again.

Maybe switching to a sqlite file will be an better option, then the initial migrations stay in place after dropping the connections.

Reply:
This should only be done in the specs necessary and not globally.